cmd/ci-secret-bootstrap/main_test.go

```diff
@@ -2633,3 +2634,621 @@ func (f *fakeVaultClient) ListKVRecursively(prefix string) ([]string, error) {
+						data:  []byte("some-name"),
+					},
+				},
+			},
+			expectedSecrets: map[string][]coreapi.Secret{
+				"cluster-1": {
+					{
+						ObjectMeta: metav1.ObjectMeta{Name: "prod-secret-1", Namespace: "namespace-1", Labels: map[string]string{"dptp.openshift.io/requester": "ci-secret-bootstrap"}},
+						Data:       map[string][]byte{"key-name-1": []byte("secret-data")},
+						Type:       coreapi.SecretTypeOpaque,
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{Name: "some-name", Namespace: "namespace-1", Labels: map[string]string{"dptp.openshift.io/requester": "ci-secret-bootstrap"}},
+						Data: map[string][]byte{
+							"some-data-key":                []byte("a-secret"),
+							"secretsync-vault-source-path": []byte("secret/user-item-1"),
+						},
+						Type: coreapi.SecretTypeOpaque,
+					},
+				},
+				"cluster-2": {
+					{
+						ObjectMeta: metav1.ObjectMeta{Name: "some-name", Namespace: "namespace-1", Labels: map[string]string{"dptp.openshift.io/requester": "ci-secret-bootstrap"}},
+						Data: map[string][]byte{
+							"some-data-key":                []byte("a-secret"),
+							"secretsync-vault-source-path": []byte("secret/user-item-1"),
+						},
+						Type: coreapi.SecretTypeOpaque,
+					},
+				},
+			},
+		},
+		{
+			id:        "Successfully merge secret from config and vault user secret in multiple clusters",
+			mergeCase: true,
+			force:     true,
+			config: secretbootstrap.Config{
+				UserSecretsTargetClusters: []string{"cluster-1", "cluster-2"},
+				Secrets: []secretbootstrap.SecretConfig{
+					{
+						From: map[string]secretbootstrap.BitWardenContext{
+							"key-name-1": {
+								BWItem: "item-name-1",
+								Field:  "field-name-1",
+							},
+						},
+						To: []secretbootstrap.SecretContext{
+							{
+								Cluster:   "cluster-1",
+								Namespace: "namespace-1",
+								Name:      "prod-secret-1",
+							},
+						},
+					},
+				},
+			},
+			secretGetters: map[string]coreclientset.SecretsGetter{
+				"cluster-1": fake.NewSimpleClientset().CoreV1(),
+				"cluster-2": fake.NewSimpleClientset().CoreV1(),
+			},
+			vaultObjects: map[string][]vaultObject{
+				"item-name-1": {
+					{
+						field: "field-name-1",
+						data:  []byte("secret-data"),
+					},
+				},
+				"user-item-1": {
+					{
+						field: "some-data-key",
+						data:  []byte("a-secret"),
+					},
+					{
+						field: "secretsync/target-namespace",
+						data:  []byte("namespace-1"),
+					},
+					{
+						field: "secretsync/target-name",
+						data:  []byte("some-name"),
+					},
+				},
+			},
+			expectedSecrets: map[string][]coreapi.Secret{
+				"cluster-1": {
+					{
+						ObjectMeta: metav1.ObjectMeta{Name: "prod-secret-1", Namespace: "namespace-1", Labels: map[string]string{"dptp.openshift.io/requester": "ci-secret-bootstrap"}},
+						Data: map[string][]byte{
+							"key-name-1":    []byte("secret-data"),
+							"merge-with-me": []byte("old-secret-data"),
```

Comment:
I'm lost - where does this come from?

Reply:
https://github.com/openshift/ci-tools/pull/1984/files#diff-5467c910200007c00123d9364d9d40c443678717319d06b6586374b03aaf8f18R3206

Reply:
addressed.